lib/dry/system/settings.rb

```diff
@@ -45,6 +46,10 @@ def self.load(root, env)
           end
 
           new(attributes)
+        rescue Dry::Struct::Error => e
+          # We remove [Configuration.new] since is an internal class and could mislead users
+          part_to_avoid = (e.message.index(']')+2)...-1
```

Comment:
do you have a full string with error? Maybe we can create a easier way to trim it 🤔

Reply:
Yeah, this does feel like a brittle way to get a nicer error message out.

For reference, here's the full `e.message` in this context:

```
[Configuration.new] "foo" (String) has invalid type for :integer_value violates constraints (type?(Integer, "foo") failed)
```

Since we own dry-struct, perhaps we can update `Dry::Struct::Error` to carry some nicer data for us to use?

I would imagine that we could add some separate attributes there for the key name and type failure message?

_Finally,_ if we're looking at better handling of these settings type errors, there's one other thing I'd like us to do that I think will make this experience a lot nicer: try applying _all_ the settings and collecting _all_ errors that occur as a part of this process, so if there's e.g. 3 missing or invalid values in someone's config, they don't have to check 3 times to fix them all up. Instead, they should see a full list of all the failing settings.

I think this would involve doing something a little different than just a straight-up `new(attributes)`, e.g. actually checking each key/val against the Configuration class' schema, but I think it'd make for a **dramatically** improved experience, which I think is an important thing for something like a dry-system application's settings.

@GustavoCaso @davydovanton What do you think?

@GustavoCaso Would you be interested in giving this a go? I figure now's a good time if you already have your head in this part of the code.

Reply:
I'll see if we can improve `Dry::Struct::Error` to see if we can provide better data for use.

And the last suggestion I think is great and would not require a huge code change in order to apply it. 

Thanks for comments @timriley and @davydovanton